### PR TITLE
Actuator integration

### PR DESCRIPTION
**Pull request by courtin**
Integrates electric motor and actuator disk propeller into the solar model.  Replaces current motor power-based sizing and constant efficiency model.  Dependent on pull request https://github.com/convexengineering/gplibrary/pull/161 to gplibrary.

### CODE REVIEW

**Comment by mjburton11** [comment]
Looks like this is just failing because convexengineering/gplibrary#161 hasn't merged yet? @courtin

File: solar/solar.py [hunk 1]
```diff
@@ -22,6 +22,8 @@
 from gpkitmodels import g
 from gpfit.fit_constraintset import FitCS as FCS
 from gpkitmodels.GP.materials import cfrpud, cfrpfabric, foamhd
+from gpkitmodels.GP.aircraft.prop.propeller import Propeller, ActuatorProp#, SimpleQProp
```
Comment (mjburton11) [inline solar/solar.py]: Delete #SimpleQProp for now?

File: solar/solar.py [hunk 2]
```diff
@@ -113,9 +117,12 @@ def setup(self, static, state):
             dvars.append(cdfuse*Sfuse/Sw)
 
         constraints = [cda >= sum(dvars),
+                       self.propeller.T==T,
```
Comment (mjburton11) [inline solar/solar.py]: Can you get rid of all the `self.` in the constraints list and make them separate variables like I do for `CL`?

File: solar/solar.py [hunk 3]
```diff
@@ -541,32 +519,35 @@ def setup(self, N, aircraft):
...
             T >= 0.5*rho*V**2*CD*S + Wtotal*hdot/V,
             hdot >= dh/dt,
             t >= sum(dt),
-            Pshaft >= T*V/etaprop,
-            E >= sum(Poper*dt)
+            E >= sum(Poper*dt),
             ]
 
-        return self.drag, constraints
+        return self.drag, constraints,# self.propeller
```
Comment (mjburton11) [inline solar/solar.py]: delete commented out part

File: solar/solar.py [hunk 4]
```diff
@@ -579,11 +560,14 @@ def setup(self, state, aircraft, perf):
         S = self.S = aircraft.wing.planform.S
         rho = self.rho = state.rho
         V = self.V = state.V
+        #self.propeller = aircraft.propeller.flight_model(aircraft.propeller, state)
```
Comment (mjburton11) [inline solar/solar.py]: delete commented out part

File: solar/solar.py [hunk 5]
```diff
@@ -579,11 +560,14 @@ def setup(self, state, aircraft, perf):
...
+        #self.propeller.substitutions['omega'] = 1000
+
 
         return [Wtotal <= (0.5*rho*V**2*CL*S),
                 T >= 0.5*rho*V**2*CD*S,
-                Pshaft >= T*V/etaprop]
-
+                perf.drag.T == T,
```
Comment (mjburton11) [inline solar/solar.py]: I don't think `T` should be created here any more.  Just have 1 `T`

**Comment by courtin** [comment]
@mjburton updated based on comments

**Comment by galbramc** [comment]
this is failing because I'm updating the VM's (or at least that might be part of it). Hopefully fixed soon.

**Comment by courtin** [comment]
@galbramc , @mjburton11 is correct this won't pass tests until convexengineering/gplibrary#161 is merged

File: solar/solar.py [hunk 6]
```diff
@@ -80,13 +83,15 @@ def setup(self, static, state):
 
         fd = dirname(abspath(__file__)) + sep + "dai1336a.csv"
 
-        self.wing = static.wing.flight_model(static.wing, state, fitdata=fd)
-        self.htail = static.emp.htail.flight_model(static.emp.htail, state)
-        self.vtail = static.emp.vtail.flight_model(static.emp.vtail, state)
-        self.tailboom = static.emp.tailboom.flight_model(static.emp.tailboom,
+        self.wing       = static.wing.flight_model(static.wing, state, fitdata=fd)
```
Comment (mjburton11) [inline solar/solar.py]: Sorry, using my formatting rules haha.  One one space between = sign please!!!

File: solar/solar.py [hunk 7]
```diff
@@ -579,11 +564,12 @@ def setup(self, state, aircraft, perf):
...
         rho = self.rho = state.rho
         V = self.V = state.V
+ 
 
-        return [Wtotal <= (0.5*rho*V**2*CL*S),
-                T >= 0.5*rho*V**2*CD*S,
-                Pshaft >= T*V/etaprop]
 
+        return [Wtotal <= (0.5*rho*V**2*CL*S),
+                perf.drag.T >= 0.5*rho*V**2*CD*S
```
Comment (mjburton11) [inline solar/solar.py]: Can you make a `T = perf.drag.T` variable above so this can read `T >= ...`?

**Comment by courtin** [comment]
test this please

**Comment by courtin** [comment]
test this please

**Comment by courtin** [comment]
test this please

**Comment by galbramc** [comment]
This error seems to be pretty persistent...

```
Traceback (most recent call last):
  File "/Users/jenkins/workspace/gpkit_ResearchModel_solar_PR/mosek/gpkit/gpkit/tests/helpers.py", line 62, in test
    testfn(name, import_dict, path)(self)
  File "/Users/jenkins/workspace/gpkit_ResearchModel_solar_PR/mosek/gpkit/gpkit/tests/from_paths.py", line 50, in <lambda>
    lambda self: getattr(self, name)()))  # pylint:disable=undefined-variable
  File "/Users/jenkins/workspace/gpkit_ResearchModel_solar_PR/mosek/gpkit/gpkit/tests/from_paths.py", line 36, in test_fn
    mod = __import__(os.path.basename(path)[:-3])
  File "season.py", line 6, in <module>
  File "solar.py", line 26, in <module>
ImportError: No module named motor.motor
```

**Comment by mjburton11** [comment]
@courtin I think the `gplibrary/gpkitmodels/GP/aircraft/motor` folder needs a `__init__.py` file.  That should solve this issue

**Comment by courtin** [comment]
test this please

**Pull request merged**